Toshi/Source/Toshi2/T2Map.h

```diff
@@ -25,21 +25,21 @@ namespace Toshi {
 			T2RedBlackTreeNode<Pair>* result = TNULL;
 			T2RedBlackTree<Pair>::Find(result, { key });
 
-			return result ? &result->GetValue()->GetSecond() : TNULL;
+			return result ? &result->GetValue()->GetSecond() : &m_RedBlackTree.End()->GetSecond();
 		}
 
-		//T2RedBlackTree<Pair>::Iterator Begin()
-		//{
-		//	return m_RedBlackTree.Begin();
-		//}
+		ValueType* Begin()
+		{
+			return &m_RedBlackTree.Begin()->GetSecond();
+		}
 
-		//T2RedBlackTree<Pair>::Iterator End()
-		//{
-		//	return m_RedBlackTree.End();
-		//}
+		ValueType* End()
+		{
+			return &m_RedBlackTree.End()->GetSecond();
+		}
 
 	private:
-		//T2RedBlackTree<Pair> m_RedBlackTree;
+		T2RedBlackTree<Pair> m_RedBlackTree;
```

Comment:
Do we need this? Cause it inherits already redblacktree

Reply:
oh right. I didn't notice you made it inherit from TRedBlackTree

Reply:
but at the same time. should it really be inherited?

Reply:
It's weird cause T2Map doesn't have any Begin() symbols in globs. Only for T2Redblacktree<Pair>. Which suggests that T2Map might inherit T2RedBlackTree

Reply:
It could be inlined and the debug info file may not store info about inline methods especially when they are coming from template class

Reply:
Well, i guess skip inheritance. We can't be sure and both works fine so let's keep it how you originally visioned